sdk/synapse/azure-analytics-synapse-spark/src/test/java/com/azure/analytics/synapse/spark/SparkSessionAsyncClientTest.java

```diff
@@ -0,0 +1,164 @@
+            .doOnNext(expected -> StepVerifier.create(client.getSparkSession(expected.getId()))
+                .assertNext(actual -> assertSparkSessionEquals(expected, actual))
+                .verifyComplete());
+    }
+
+    @Test
+    public void crudSparkSession() {
+        // arrange
+        String sessionName = testResourceNamer.randomName("spark-session-", 20);
+        SparkSessionOptions options = new SparkSessionOptions()
+            .setName(sessionName)
+            .setDriverMemory("28g")
+            .setDriverCores(4)
+            .setExecutorMemory("28g")
+            .setExecutorCores(4)
+            .setExecutorCount(2);
+
+        AtomicReference<SparkSession> testSession = new AtomicReference<>();
+
+        try {
+            // act
+            StepVerifier.create(client.createSparkSession(options, true))
+                .consumeNextWith(expected -> {
+                    testSession.set(expected);
+
+                    assertEquals(sessionName, expected.getName());
+                    assertEquals(getSparkPoolName(), expected.getSparkPoolName());
+                })
+                .verifyComplete();
+
+            // act
+            StepVerifier.create(client.getSparkSession(testSession.get().getId(), true))
+                .assertNext(actual -> assertSparkSessionEquals(testSession.get(), actual))
+                .verifyComplete();
+        } finally {
+            // clean up
+            if (testSession.get() != null) {
+                client.cancelSparkSession(testSession.get().getId()).block();
+            }
+        }
+    }
+
+    @Test
+    public void crudSparkStatement() throws Exception {
+        // arrange
+        String sessionName = testResourceNamer.randomName("spark-session-", 20);
+        SparkSessionOptions sessionOptions = new SparkSessionOptions()
+            .setName(sessionName)
+            .setDriverMemory("28g")
+            .setDriverCores(4)
+            .setExecutorMemory("28g")
+            .setExecutorCores(4)
+            .setExecutorCount(2);
+
+        AtomicReference<SparkSession> testSession = new AtomicReference<>();
+        AtomicReference<SparkStatement> testStatement = new AtomicReference<>();
+
+        try {
+            // act
+            StepVerifier.create(client.createSparkSession(sessionOptions, true))
+                .consumeNextWith(session -> {
+                    testSession.set(session);
+
+                    // assert
+                    assertEquals(sessionName, session.getName());
+                    assertEquals(getSparkPoolName(), session.getSparkPoolName());
+                })
+                .verifyComplete();
+
+            // arrange
+            String code = "print('hello, Azure CLI')";
+
+            SparkStatementOptions options = new SparkStatementOptions()
+                .setKind(SparkStatementLanguageType.PYSPARK)
+                .setCode(code);
+
+            Mono<SparkStatement> createSparkStatement = Mono.defer(() -> {
```

Comment:
I'm assuming this is the code you referred to as needing to be LRO'ed?

Reply:
This I'm not sure. I did exactly the same as the tests in CLI to wait for 6 minutes. The operations themselves are not marked with `x-ms-long-running-operation`: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/sparkJob.json#L462. They might be - but I'm not too sure until I understand more of the [standard LRO patterns](https://github.com/Microsoft/api-guidelines/blob/master/Guidelines.md#13-long-running-operations).